addons/website_hr_recruitment/models/hr_recruitment.py

```diff
@@ -30,15 +31,17 @@ class Applicant(models.Model):
     _inherit = 'hr.applicant'
 
     def website_form_input_filter(self, request, values):
+        job = self.env['hr.job'].browse(values.get('job_id'))
+        if not job or not job.sudo().website_published:
+            raise UserError(_("You cannot apply for this job."))
```

Comment:
Seems violent for stable, what about people with form to apply for a department only, without a job?